scripts/cc-cli/src/utils/tx.ts

```diff
@@ -0,0 +1,67 @@
+import { ISubmittableResult } from "@polkadot/types/types";
+import { ApiPromise, KeyringPair } from "creditcoin-js";
+
+import { SubmittableExtrinsic } from "@polkadot/api/types";
+
+export async function signSendAndWatch(
+  tx: SubmittableExtrinsic<"promise", ISubmittableResult>,
+  api: ApiPromise,
+  signer: KeyringPair
+): Promise<TxResult> {
+  // Top-Level Promise
+
+  // eslint-disable-next-line @typescript-eslint/no-misused-promises
+  return new Promise(async (resolve) => {
+    // eslint-disable-next-line @typescript-eslint/no-misused-promises
+    const out: TxResult = await new Promise(async (resolveInner) => {
+      console.log("Sending transaction...");
+      // Sign and send with callback
+      await tx.signAndSend(signer, ({ status, dispatchError }) => {
+        // Called every time the status changes
+        if (status.isFinalized) {
+          const result = {
+            status: TxStatus.ok,
+            info: `Transaction included at blockHash ${status.asFinalized.toString()}`,
+          };
+
+          resolveInner(result);
+        }
+        if (dispatchError) {
+          if (dispatchError.isModule) {
+            // for module errors, the section is indexed, lookup
+            const decoded = api.registry.findMetaError(dispatchError.asModule);
+            const { docs, name, section } = decoded;
+
+            const error = `${section}.${name}: ${docs.join(" ")}`;
+
+            const result = {
+              status: TxStatus.failed,
+              info: `Transaction failed with error: "${error}"`,
+            };
+
+            resolveInner(result);
+          } else {
+            // Other, CannotLookup, BadOrigin, no extra info
+            const result = {
+              status: TxStatus.failed,
+              info: `Transaction failed with error: "${dispatchError.toString()}"`,
+            };
+            resolveInner(result);
+          }
+        }
+      });
+    });
+    resolve(out);
+  });
```

Comment:
```suggestion
  return new Promise((resolve) => {
    console.log("Sending transaction...");
    let maybeUnsub: (() => void) | undefined;
    const unsubAndResolve = (result: TxResult) => {
      if (maybeUnsub) {
        maybeUnsub();
      }
      resolve(result);
    };
    // Sign and send with callback
    tx.signAndSend(signer, ({ status, dispatchError }) => {
      // Called every time the status changes
      if (status.isFinalized) {
        const result = {
          status: TxStatus.ok,
          info: `Transaction included at blockHash ${status.asFinalized.toString()}`,
        };
        unsubAndResolve(result);
      }
      if (dispatchError) {
        if (dispatchError.isModule) {
          // for module errors, the section is indexed, lookup
          const decoded = api.registry.findMetaError(dispatchError.asModule);
          const { docs, name, section } = decoded;

          const error = `${section}.${name}: ${docs.join(" ")}`;

          const result = {
            status: TxStatus.failed,
            info: `Transaction failed with error: "${error}"`,
          };

          unsubAndResolve(result);
        } else {
          // Other, CannotLookup, BadOrigin, no extra info
          const result = {
            status: TxStatus.failed,
            info: `Transaction failed with error: "${dispatchError.toString()}"`,
          };
          unsubAndResolve(result);
        }
      }
    }).then((unsub) => {
      maybeUnsub = unsub;
    });
  });
```

Reply:
Github's diff is a little rough bc of the indentation change, here's a better diff:
![Screenshot 2023-06-21 at 3 28 49 PM](https://github.com/gluwa/creditcoin/assets/17734409/af351933-3257-4c24-8ef6-65eff0d321f5)